File: objectorientedprmng/inheritance.py
#single inheritance
class parent:
    def phone(self):
        print("inside parents phone method")
class child(parent):
    pass
ch=child()
ch.phone()


#multilevel inheritance
class parent:
    def m1(self):
        print("inside m1")
class SubChild(parent):
    def m2(self):
        print("inside m2")
class Child(SubChild):
    def m3(self):
        print("inside m3")
ch=Child()
ch.m1()
ch.m2()
ch.m3()


#multiple inheritance
class parent:
    def m1(self):
        print("m1")
class SubChild:
    def m2(self):
        print("m2")
class Child(SubChild,parent):
    def m3(self):
        print("m3")
ch=Child()
ch.m1()
ch.m2()
ch.m3()

File: htmlportions/JavaScript/objectorientedprgmng/inheritance.js
//inheritance

// class parent{
//     phone()
//     {
//         console.log("have nokia");
//     }
// }
// class child extends parent{

// }
// var ch=new child()
// ch.phone()


//multilevel

class parent{
    m1(){
        console.log("have nokia");
    }
}
class SubChild extends parent{
    m2(){
        console.log("m2");
        }
}
class Child extends SubChild{
    m3(){
        console.log("m3");
    }
}

var ch=new Child()
ch.m3()
ch.m2()
ch.m1()